### PR TITLE
add nginx exporter

### PR DESCRIPTION
**Pull request by kubicgruenfeld**
#### Pull Request (PR) description
This PR adds the nginx-prometheus-exporter from nginx inc.

#### This Pull Request (PR) fixes the following issues
n/a

### CODE REVIEW

File: data/defaults.yaml [hunk 1]
```diff
@@ -320,3 +320,12 @@ prometheus::grok_exporter::package_ensure: 'latest'
 prometheus::grok_exporter::package_name: 'grok_exporter'
 prometheus::grok_exporter::user: 'grok-exporter'
 prometheus::grok_exporter::version: '1.0.0.RC4'
+prometheus::nginx_prometheus_exporter::scrape_uri: 'http://localhost:8080/stub_status'
```
Comment (bastelfreak) [inline data/defaults.yaml]: can you please place the static data in the nginx_prometheus_exporter.pp instead of the defaults.yaml? that enables puppet-strings to display it in the REFERENCE.md. We want to move the other data away as well, but lacking time at the moment.

Comment (kubicgruenfeld) [reply data/defaults.yaml]: sure

**Comment by kubicgruenfeld** [comment]
i got stuck on the acceptance tests. i think the problem is, that the exporter exits, if the status page is not reachable and i don't know how to proceed here.

File: metadata.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
   "dependencies": [
     {
       "name": "puppetlabs/stdlib",
-      "version_requirement": ">= 4.25.0 < 9.0.0"
+      "version_requirement": ">= 4.25.0 < 8.0.0"
```
Comment (root-expert) [inline metadata.json]: This seems unrelated and should be reverted

File: spec/acceptance/nginx_prometheus_exporter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  describe 'default install' do
+    describe service('nginx_prometheus_exporter') do
+      it { is_expected.to be_running }
+      it { is_expected.to be_enabled }
+    end
+    describe port(9113) do
+      it { is_expected.to be_listening.with('tcp6') }
+    end
+    describe process('nginx-prometheus-exporter') do
+      its(:args) { is_expected.to match %r{\ -nginx.scrape-uri http://localhost:8888/stub_status} }
```
Comment (root-expert) [inline spec/acceptance/nginx_prometheus_exporter_spec.rb]: Maybe this will fix the failing spec?

```suggestion
      its(:args) { is_expected.to match %r{-nginx\.scrape-uri 'http:\/\/localhost:8888\/stub_status'} }
```

File: metadata.json (hunk 2 above)
Comment (kubicgruenfeld) [reply metadata.json]: it was just for testing, because without the downgrade i get:

```
  Error: Could not install module 'puppet-nginx' (v3.2.0)
    The requested version cannot satisfy one or more of the following installed modules:
      puppetlabs-stdlib, installed: 8.0.0, expected: >= 5.0.0 < 8.0.0
```

this will be fixed by a new release of puppet-nginx

File: spec/acceptance/nginx_prometheus_exporter_spec.rb (hunk 3 above)
Comment (kubicgruenfeld) [reply spec/acceptance/nginx_prometheus_exporter_spec.rb]: actually it was a bit weirder.

it seems the `describe process` is executing `/bin/sh -c ps -C nginx-prometheus-exporter -o args= | head -1` and in newer versions of ps the cmd gets truncated to 15 characters.

File: .fixtures.yml [hunk 4]
```diff
@@ -1,6 +1,8 @@
 fixtures:
   repositories:
+    apt: "https://github.com/puppetlabs/puppetlabs-apt"
```
Comment (bastelfreak) [inline .fixtures.yml]: are the changes in the .fixtures.yml actually required? the exporter doesn't use apt/nginx, only the acceptance tests. but the fixtures.yml is only used within unit tests. a possible way to install modules in acceptance tests is this: https://github.com/voxpupuli/puppet-grafana/blob/master/spec/acceptance/class_spec.rb#L8-L10

Comment (kubicgruenfeld) [reply .fixtures.yml]: ah, didn't know this.

the modules are installed in `spec/acceptance/nginx_prometheus_exporter_spec.rb` with:

```
describe 'prometheus nginx exporter' do
  it 'nginx_prometheus_exporter works idempotently with no errors' do
    on hosts, puppet('module', 'install', 'puppet-nginx', '--version', '3.2.1')
    on hosts, puppet('module', 'install', 'puppetlabs-apt', '--version', '8.2.0')
```

File: spec/acceptance/nginx_prometheus_exporter_spec.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
+require 'spec_helper_acceptance'
+
+describe 'prometheus nginx exporter' do
+  it 'nginx_prometheus_exporter works idempotently with no errors' do
+    on hosts, puppet('module', 'install', 'puppet-nginx', '--version', '3.2.1')
```
Comment (bastelfreak) [inline spec/acceptance/nginx_prometheus_exporter_spec.rb]: can you use the example from the grafana module I linked? That's better because it doesn't hardcode a specific version

Comment (kubicgruenfeld) [reply spec/acceptance/nginx_prometheus_exporter_spec.rb]: done

**Comment by bastelfreak** [comment]
thanks for all the work!
